### PR TITLE
Tentative fix for saving records with new workflow

### PR DESCRIPTION
**Pull request by NightJar**
Still breaks on subsequent save & publish, but the first one works

Stop gap fix for #333

### CODE REVIEW

File: src/Extensions/WorkflowApplicable.php [hunk 1]
```diff
@@ -319,6 +327,19 @@ public function LinkToPendingItems()
      */
     public function onAfterWrite()
     {
+        // Record changes to the workflow on this record
+        $changed = $this->owner->getChangedFields(['WorkflowDefinitionID']);
+        if (isset($changed['WorkflowDefinitionID']) &&
+            (
+                empty($changed['WorkflowDefinitionID']['before'])
+                || empty($changed['WorkflowDefinitionID']['before'])
```
Comment (robbieaverill) [inline src/Extensions/WorkflowApplicable.php]: Are these the same thing?

File: src/Extensions/WorkflowApplicable.php [hunk 2]
```diff
@@ -382,6 +403,12 @@ public function RecentWorkflowComment($limit = 10)
      */
     public function canPublish()
     {
+        // Ignore canPublish() if in the middle of having workflow added / removed
+        $key = $this->owner->baseClass() . '#' . $this->owner->ID;
+        if (isset(static::$itemsChangingWorkflow[$key])) {
+            return null;
```
Comment (robbieaverill) [inline src/Extensions/WorkflowApplicable.php]: I feel like `false` would suit the method name better, assuming it should return a bool (unless false and null have different results for the user, like a validation error)

File: src/Extensions/WorkflowApplicable.php (hunk 1 above)
Comment (NightJar) [reply src/Extensions/WorkflowApplicable.php]: It appears to be, yep. Perhaps the first was supposed to be `isset`.

Comment (robbieaverill) [reply src/Extensions/WorkflowApplicable.php]: `isset` would be covered by `empty` anyway, maybe we just need the one

File: src/Extensions/WorkflowApplicable.php (hunk 2 above)
Comment (raissanorth) [reply src/Extensions/WorkflowApplicable.php]: They do, and this has been addressed in #345

**Comment by raissanorth** [comment]
This pull request is now superseded by #345 
Closing.

**Pull request closed without merging**